framework/src/modules/random/utils.ts

```diff
@@ -67,14 +67,33 @@ export const getRandomSeed = (
 	numberOfSeeds: number,
 	validatorsReveal: ValidatorSeedReveal[],
 ) => {
+	if (!Number.isInteger(height) || !Number.isInteger(numberOfSeeds)) {
+		throw new Error('Height or number of seeds cannot be non integer.');
+	}
 	if (height < 0 || numberOfSeeds < 0) {
 		throw new Error('Height or number of seeds cannot be negative.');
 	}
+	if (numberOfSeeds > 1000) {
+		throw new Error('Number of seeds cannot be greater than 1000.');
+	}
 	const initRandomBuffer = utils.intToBuffer(height + numberOfSeeds, 4);
 	let randomSeed = cryptography.utils.hash(initRandomBuffer).slice(0, 16);
-	const currentSeeds = validatorsReveal.filter(
-		v => height <= v.height && v.height <= height + numberOfSeeds,
-	);
+
+	let isInFuture = true;
+	const currentSeeds = [];
+	for (const validatorReveal of validatorsReveal) {
+		if (validatorReveal.height >= height) {
+			isInFuture = false;
```

Comment:
in the original code, `validatorReveal.height == height` is also selected without error, and equal is not in the future i thinks

Reply:
equal is still not in future. 
`if(validatorReveal.height === height)` then `isInFuture = false` and it does not throw error. It throws error when `isInFuture = true`

Reply:
ah that's right. with this logic, if at least one is not in the future it will not throw error, but is that how it's intended?

Reply:
From the issue description, apparently yes
```
if for every seedObject element in validatorReveals height > seedObject.height
    Expectation: fail 
```

Reply:
ah ok, then the logic is correct. it took me a while to understand why this is ok.

For my understanding, it was like
```
height <= seedObject.height < height + numberOfSeed
```
the seed is selected like above and if there is no seedObject below specified height, the `height` input is incorrect
Also, in the code, `validatorReveal` is the `seedObject`.